webapp/src/components/gallery/galleryCard.tsx

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useState, useEffect} from 'react'
+import {FormattedMessage} from 'react-intl'
+
+import {Card} from '../../blocks/card'
+import {CardTree, MutableCardTree} from '../../viewModel/cardTree'
+import {IContentBlock} from '../../blocks/contentBlock'
+
+import useCardListener from '../../hooks/cardListener'
+
+import ImageElement from '../content/imageElement'
+import ContentElement from '../content/contentElement'
+
+import './galleryCard.scss'
+
+type Props = {
+    card: Card
+    showCard: (cardId: string) => void
+}
+
+const GalleryCard = React.memo((props: Props) => {
+    const {card} = props
+    const [cardTree, setCardTree] = useState<CardTree>()
+    useCardListener(
+        card.id,
+        async (blocks) => {
+            const newCardTree = cardTree ? MutableCardTree.incrementalUpdate(cardTree, blocks) : await MutableCardTree.sync(card.id)
+            setCardTree(newCardTree)
+        },
+        async () => {
+            const newCardTree = await MutableCardTree.sync(card.id)
```

Comment:
This is fine for now, but some thoughts on how we could reduce the number of API calls in the future:
* Let the Gallery own the state
* Initially we could modify / extend BoardTree to call getSubTree with 3 levels
* I envision adding a `getTree` API that returns all nodes by rootID
* And updating the websocket to notify changes to trees by rootID as well
* Then state can be managed by a `fullBoardTree` view model
* In the future-future, we can look at adding filters to the query (e.g. to limit types of nodes), or build something closer to GraphQL.

Just initial thoughts.